src/main/java/org/ibp/api/java/impl/middleware/study/validator/StudyImportRequestValidator.java

```diff
@@ -0,0 +1,129 @@
+package org.ibp.api.java.impl.middleware.study.validator;
+
+import org.apache.commons.lang3.StringUtils;
+import org.generationcp.middleware.api.brapi.v2.study.StudyImportRequestDTO;
+import org.generationcp.middleware.api.location.LocationService;
+import org.generationcp.middleware.api.location.search.LocationSearchRequest;
+import org.generationcp.middleware.domain.dms.StudySummary;
+import org.generationcp.middleware.service.api.study.StudySearchFilter;
+import org.generationcp.middleware.service.api.study.StudyService;
+import org.ibp.api.java.impl.middleware.common.validator.BaseValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.MapBindingResult;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+
+@Component
+public class StudyImportRequestValidator {
+
+	@Autowired
+	private StudyService studyService;
+
+	@Autowired
+	private LocationService locationService;
+
+	protected BindingResult errors;
+
+	public BindingResult pruneStudiesInvalidForImport(final List<StudyImportRequestDTO> studyImportRequestDTOS, final String crop) {
+		BaseValidator.checkNotEmpty(studyImportRequestDTOS, "study.import.request.null");
+		this.errors = new MapBindingResult(new HashMap<>(), StudyImportRequestDTO.class.getName());
+
+		Integer index = 1;
+		final Iterator<StudyImportRequestDTO> iterator = studyImportRequestDTOS.iterator();
+		while (iterator.hasNext()) {
+			final StudyImportRequestDTO s = iterator.next();
+			if (StringUtils.isEmpty(s.getTrialDbId())) {
+				this.errors.reject("study.import.trialDbId.required", new String[] {index.toString()}, "");
+				iterator.remove();
+				continue;
+			}
+
+			final StudySearchFilter filter = new StudySearchFilter();
+			filter.setTrialDbIds(Collections.singletonList(s.getTrialDbId()));
+			final List<StudySummary> studies = this.studyService.getStudies(filter, null);
```

Comment:
Can we call getStudies once for all trialDbIds (maybe build a map by trialDbId)?